### ISSUE TITLE
About the setting of evaluation

### ISSUE THREAD

**Issue by workerbcd** [leokarlin/BrAD#4]
Thank you for your impressive work,

I am a new researcher in the field of unsupervised learning. I have a question about the setting of the evaluation part. I notice that the classifier is fit with source labels. Does this break the initial purpose of unsupervised learning? 
https://github.com/leokarlin/BrAD/blob/1fbb23f6ce602e2accdd207710592994efc565a8/main_brad_test.py#L182-L193

Look forward to your reply. 
Regards,

**Comment by EliSchwartz** [comment]
This is the test script. A small subset of the source samples is labeled and used for fitting a classifier after the pre-training is done. Note that we don't use these labels at the pre-training stage.

**Comment by workerbcd** [comment]
Thank you for your reply,

We evaluate your released pretrained model on DomainNet on the task of UDG. The testing command is in the following:

_#!/bin/bash
data_root='/home/xxx/DomainNet'
exp_root='/home/xxx/Bradlogs'
name=oracle

python -m torch.distributed.launch --nproc_per_node=2 main_test.py -a resnet18 \
--resume $exp_root/$name/UDG_domainnet_source-CIQ.pth.tar \
--src-domain $data_root/real_test.txt --dst-domain $data_root/real_test.txt \
--classifier retrieval_

To be noticed that the real_test.txt file is achieved from the official website: http://ai.bu.edu/M3SDA/

And here is the result we get(label fraction is 0.01 and the label choice is random) :

_Results:
acc1=0.0870791468358893,acc10=0.24502027146130795,acc20=0.3251806804160056,precision1=0.0870791468358893,precision5=0.043213467301251564,precision10=0.030768552793936203,precision15=0.02515130148657382,precision20=0.021749955931605856_

I think maybe there is something wrong when we implement the evaluation part since the result we get is far away different from the one in your paper. Could you release the testing code on UDG to help us check our misunderstanding?

Thank you very much and look forward to your reply,
Regards,
